flink-streaming-java/src/main/java/org/apache/flink/streaming/api/transformations/AbstractMultipleInputTransformation.java

```diff
@@ -76,10 +78,10 @@ public StreamOperatorFactory<OUT> getOperatorFactory() {
     }
 
     @Override
-    public List<Transformation<?>> getTransitivePredecessors() {
```

Comment:
A simpler way can be

```
        return inputs.stream()
                .flatMap(input -> input.getTransitivePredecessors().stream())
                .distinct();
                .collect(Collectors.toList());
```

Reply:
Well. I just noticed a potential bug that the transformation itself is not added to the predecessors, unlike all other transformations.